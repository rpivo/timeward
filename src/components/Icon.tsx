import React from 'react';
import { iconPaths } from '@utilities/svg';

type IconProps = {
  type?: string;
};

const Checkmark: React.FC = (): JSX.Element =>
  <svg width='299.106' height='299.106' viewBox='0 0 299.106 299.106'>
    <path
      d='M113.5,249v-.5H0v-60H113.5V0h60V188.5h.5v60h-.5v.5Z'
      transform='translate(176.07) rotate(45)'
      fill='#00C519'
    />
  </svg>;

const IconX: React.FC = (): JSX.Element =>
  <svg
    style={{ marginBottom: '-0.15rem', marginTop: '0.15rem' }}
    width='218.496'
    height='218.496'
    viewBox='0 0 218.496 218.496'
  >
    <path d={iconPaths.iconX} fill='red' />
  </svg>;

const Icon: React.FC<IconProps> = ({ type }: IconProps): JSX.Element => {
  switch (type) {
    case 'x':
      return <IconX />;
    case 'checkmark':
    default:
      return <Checkmark />;
  }
};

export default Icon;
