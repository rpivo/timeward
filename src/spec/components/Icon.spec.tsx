import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '@components/Icon';

describe('Icon', () => {
  it('should render X icon', () => {
    const tree = renderer
      .create(<Icon type='x' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render checkmark icon', () => {
    const tree = renderer
      .create(<Icon type='checkmark' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
