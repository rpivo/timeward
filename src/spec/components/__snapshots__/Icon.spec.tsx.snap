// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<svg
  height="299.106"
  viewBox="0 0 299.106 299.106"
  width="299.106"
>
  <path
    d="M113.5,249v-.5H0v-60H113.5V0h60V188.5h.5v60h-.5v.5Z"
    fill="#00C519"
    transform="translate(176.07) rotate(45)"
  />
</svg>
`;
